src/core/ad.jl

```diff
@@ -108,38 +110,20 @@ gradient2(_vi::VarInfo, model::Function, spl::Union{Nothing, Sampler}) = begin
   g(vi[spl])
 end
 
-@init @require ReverseDiff="37e2e3b7-166d-5795-8a7a-e32c996b4267" begin
-
 gradient_r(theta::Vector{Float64}, vi::VarInfo, model::Function) = gradient_r(theta, vi, model, nothing)
 gradient_r(theta::Vector{Float64}, vi::Turing.VarInfo, model::Function, spl::Union{Nothing, Sampler}) = begin
-    inputs = (theta)
-
-    if Turing.ADSAFE || (spl == nothing || length(spl.info[:reverse_diff_cache]) == 0)
-        f_r(ipts) = begin
-          vi[spl][:] = ipts[:]
-          -runmodel(model, vi, spl).logp
-        end
-        gtape = GradientTape(f_r, inputs)
-        ctape = compile(gtape)
-        res = (similar(theta))
-
-        if spl != nothing
-          spl.info[:reverse_diff_cache][:ctape] = ctape
-          spl.info[:reverse_diff_cache][:res] = res
-        end
-    else
-        ctape = spl.info[:reverse_diff_cache][:ctape]
-        res = spl.info[:reverse_diff_cache][:res]
+    f_r(ipts) = begin
```

Comment:
Might be an idea to either comment this a bit more thoroughly or change the variable names to self-document a bit better. Probably both. Alternatively, just make `f_r` an anonymous function, as it's only used in the `gradient` call: that would avoid the need to choose a name for it.

Reply:
Good idea!